src/lib/InstantSearch.ts

```diff
@@ -471,6 +471,11 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
     }
   }
 
+  public onChange = defer(() => {
```

Comment:
Not sure about `defer` for this function. The deferred function does trigger a side effect (like a network request) that's why we batch them. But this callback is a way to notify the "middleware" that something happens. IMO the decision to pick or not the change belongs to the "middleware", not InstantSearch.

Reply:
The idea was more to avoid recomputing `nextUiState` too often and avoid exposing intermediate states to the middlewares.

But since the middlewares are not really defined yet then we can keep a simple function for now and see later if it is useful to defer it.